### DISCUSSION TITLE
udpBatchSize或许可以有个更合适的逻辑

### DISCUSSION THREAD

**Discussion by dev4u** [database64128/shadowsocks-go#22] [Ideas]
udpBatchSize(ubs)这个参数能不能在建立连接的初阶段(握手、连接建立后的N秒内)允许将udp数据，不按ubs设定值转发数据？
这样主要是解决一些udp应用，可以快速完成“握手”到发送payload的过程。不至于让应用一直在傻等着握手包影响使用体验。

**Comment by database64128** [comment]
Receiving and sending UDP packets in batches via the `recvmmsg(2)` and `sendmmsg(2)` syscalls do not introduce any additional delay (unless you intentionally do so, which we don't). Packets are relayed as soon as they are received and processed. The `udpBatchSize` option is only useful for low memory systems, since the default batch size (256) is quite aggressive and will consume a lot of memory.

**Reply by dev4u** [reply]
或许我要找个时间做下对比，因为我在拨telegram语音的时候，连通的时间真的存在差距。两个号互拨，手机上已经显示接通了，并显示接通时间记数，而macos上的tg还在显示接通中。

**Reply by database64128** [reply]
UDP batch relay is only available on Linux. Setting `udpBatchSize` on other platforms has no effect.

Your experience with Telegram doesn't really mean anything. My guess is Telegram didn't use proxy for calls on your macOS device.

**Reply by dev4u** [reply]
嗯，的确我一个人觉得有问题，好像也不是什么大问题。不过tg语音肯定是走代理了，不然也不会成功建立连接。
没关系，我再观察一下吧。

**Reply by database64128** [reply]
Telegram's voice calls feature works without proxy. Unless you see in shadowsocks-go's logs that Telegram is sending UDP packets, it is more likely that it didn't use proxy for voice calls.
